wandb/sdk/launch/builder/build.py

```diff
@@ -323,8 +324,19 @@ def generate_dockerfile(
     builder_type: str,
     dockerfile: Optional[str] = None,
 ) -> str:
-    if dockerfile:
+    override_entrypoint = launch_project.override_entrypoint or entry_point
+    if not dockerfile and override_entrypoint.name is not None:
         # For non-image-based jobs, project_dir is always set
+        assert (
```

Comment:
TIOLI: this line and duplicate assertion on [L340](https://github.com/wandb/wandb/pull/6122/files#diff-c24243e202a8bfb8b51dd0d8716256835ffa58e7712edfb9098ebc737a0adfbfR340) are a smell...  These want to be an `if` wrapping those protected blocks.  i.e. value of `launch_project.project_dir` determines the behavior of this function more than the values of `dockerfile` or `override_entrypoint`, so "promote" it to be an earlier decision.  I think it would make this section L327-347 much cleaner.